packages/edit-site/src/components/page-patterns/dataviews-patterns.js

```diff
@@ -0,0 +1,339 @@
+	const isUncategorizedThemePatterns =
+		type === PATTERN_TYPES.theme && categoryId === 'uncategorized';
+	const { patterns, isResolving } = usePatterns(
+		type,
+		isUncategorizedThemePatterns ? '' : categoryId,
+		{
+			search: view.search,
+			// syncStatus:
+			// 	deferredSyncedFilter === 'all'
+			// 		? undefined
+			// 		: deferredSyncedFilter,
+		}
+	);
+	const fields = useMemo(
+		() => [
+			{
+				header: __( 'Preview' ),
+				id: 'preview',
+				render: ( { item } ) => (
+					<Preview item={ item } viewType={ view.type } />
+				),
+				minWidth: 120,
+				maxWidth: 120,
+				enableSorting: false,
+				enableHiding: false,
+			},
+			{
+				header: __( 'Title' ),
+				id: 'title',
+				getValue: ( { item } ) => item.title,
+				render: ( { item } ) => (
+					<Title
+						item={ item }
+						onClick={ () => {} }
+						categoryId={ categoryId }
+					/>
+				),
+				maxWidth: 400,
+				enableHiding: false,
+			},
+		],
+		[ view.type, categoryId ]
+	);
+
+	const { data, paginationInfo } = useMemo( () => {
+		if ( ! patterns ) {
+			return {
+				data: EMPTY_ARRAY,
+				paginationInfo: { totalItems: 0, totalPages: 0 },
+			};
+		}
+		let filteredData = [ ...patterns ];
+		// Handle filters.
+		if ( view.filters.length > 0 ) {
+			// view.filters.forEach( ( filter ) => {
+			// 	if (
+			// 		filter.field === 'author' &&
+			// 		filter.operator === OPERATOR_IN &&
+			// 		!! filter.value
+			// 	) {
+			// 		filteredData = filteredData.filter( ( item ) => {
+			// 			return item.author_text === filter.value;
+			// 		} );
+			// 	} else if (
+			// 		filter.field === 'author' &&
+			// 		filter.operator === OPERATOR_NOT_IN &&
+			// 		!! filter.value
+			// 	) {
+			// 		filteredData = filteredData.filter( ( item ) => {
+			// 			return item.author_text !== filter.value;
+			// 		} );
+			// 	}
+			// } );
+		}
+
+		// Handle sorting.
+		if ( view.sort ) {
+			const stringSortingFields = [ 'title' ];
+			const fieldId = view.sort.field;
+			if ( stringSortingFields.includes( fieldId ) ) {
+				const fieldToSort = fields.find( ( field ) => {
+					return field.id === fieldId;
+				} );
+				filteredData.sort( ( a, b ) => {
+					const valueA = fieldToSort.getValue( { item: a } ) ?? '';
+					const valueB = fieldToSort.getValue( { item: b } ) ?? '';
+					return view.sort.direction === 'asc'
+						? valueA.localeCompare( valueB )
+						: valueB.localeCompare( valueA );
+				} );
+			}
+		}
+
+		// Handle pagination.
+		const start = ( view.page - 1 ) * view.perPage;
+		const totalItems = filteredData?.length || 0;
+		filteredData = filteredData?.slice( start, start + view.perPage );
+		return {
+			data: filteredData,
+			paginationInfo: {
+				totalItems,
+				totalPages: Math.ceil( totalItems / view.perPage ),
+			},
+		};
+	}, [ patterns, view, fields ] );
+
+	const actions = useMemo( () => [ renameAction, exportJSONaction ], [] );
+	const onChangeView = useCallback(
+		( viewUpdater ) => {
+			let updatedView =
```

Comment:
The logic for the callback to receive a function can be removed, no DataViews code uses a function as of now. The dependencies for this callback can be `view.type` and `setView`, so it is not updated on every view change. See, for example, https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/page-pages/index.js#L307